types/types.go

```diff
@@ -732,16 +732,17 @@ type CredentialSpecConfig struct {
 
 // FileObjectConfig is a config type for a file used by a service
 type FileObjectConfig struct {
-	Name           string            `yaml:"name,omitempty" json:"name,omitempty"`
-	File           string            `yaml:"file,omitempty" json:"file,omitempty"`
-	Environment    string            `yaml:"environment,omitempty" json:"environment,omitempty"`
-	Content        string            `yaml:"content,omitempty" json:"content,omitempty"`
-	External       External          `yaml:"external,omitempty" json:"external,omitempty"`
-	Labels         Labels            `yaml:"labels,omitempty" json:"labels,omitempty"`
-	Driver         string            `yaml:"driver,omitempty" json:"driver,omitempty"`
-	DriverOpts     map[string]string `yaml:"driver_opts,omitempty" json:"driver_opts,omitempty"`
-	TemplateDriver string            `yaml:"template_driver,omitempty" json:"template_driver,omitempty"`
-	Extensions     Extensions        `yaml:"#extensions,inline,omitempty" json:"-"`
+	Name            string            `yaml:"name,omitempty" json:"name,omitempty"`
+	File            string            `yaml:"file,omitempty" json:"file,omitempty"`
+	Environment     string            `yaml:"environment,omitempty" json:"environment,omitempty"`
+	Content         string            `yaml:"content,omitempty" json:"content,omitempty"`
+	marshallContent bool              // configure marshalling to include Content - excluded by default to prevent sensitive data leaks
```

Comment:
Should be public, no?
```suggestion
	MarshallContent bool              // configure marshalling to include Content - excluded by default to prevent sensitive data leaks
```

Reply:
nope, so that the only way to set this is to use the mashaller option (this is just an implementation detail here, as yaml.Marshall does not allow passing options or custom marshallers)